node/pod.go

```diff
@@ -216,9 +216,26 @@ func (pc *PodController) updatePodStatus(ctx context.Context, podFromKubernetes
 	kPod.Lock()
 	podFromProvider := kPod.lastPodStatusReceivedFromProvider.DeepCopy()
 	kPod.Unlock()
-	if cmp.Equal(podFromKubernetes.Status, podFromProvider.Status) {
+	if cmp.Equal(podFromKubernetes.Status, podFromProvider.Status) && podFromProvider.DeletionTimestamp == nil {
 		return nil
 	}
+	if podFromProvider.DeletionTimestamp != nil && podFromKubernetes.DeletionTimestamp == nil {
```

Comment:
Can you add a comment as to what cases would hit this?

Reply:
Also, why not just call delete from your provider?

Reply:
Provider should not delete pod in K8s

Reply:
IIRC the VK does (used to do) this so the provider doesn't do it. When a deletionTimestamp != 0, VK must ensure all resources are deleted in the provider, respecting grace period, and ultimately delete the Pod from the API.

This is not the first time we discuss this so there's history in this repo if someone has time to go through all of it.